File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int currMax = 0;
        int l = prices[0];
        for(int i = 0; i < prices.length; i++){
            currMax = prices[i]-l;
            if(prices[i]<l){
                l = prices[i];
            }
            if (currMax > max){
                max = currMax;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/insert-new-interval/submission-0.py
class Solution:
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
        result = []
        i, n = 0, len(intervals)

        while i<n and intervals[i][1]<newInterval[0]:
            result.append(intervals[i])
            i += 1
        
        x, y = newInterval[0], newInterval[1]
        while i<n and intervals[i][0] <= y:
            x = min(intervals[i][0], x)
            y = max(intervals[i][1], y)
            i += 1
        result.append([x,y])

        while i<n:
            result.append(intervals[i])
            i+=1 
        
        return result